### ISSUE TITLE
Missing character in "ValidRegex" pattern of ComponentFieldName

### ISSUE THREAD

**Issue by claorden** [tehmantra/saunter#16]
In my class properties I have some nullable components. In AsyncApiSchema/v2/ComponentFieldName.cs there is a validation of the component field name which does not include the character "\`" of my type "nullable`1", which breaks the process.

Just adding the character to ValidRegex fixes the problem:

```
private const string ValidRegex = @"^[a-zA-Z0-9\.\-_`]+$";
```

Any collateral damage you could think of after adding this?

Thank you for your work!

**Comment by tehmantra** [comment]
This is following the AsyncApi specification. See: https://www.asyncapi.com/docs/specifications/2.0.0/#componentsObject

> All the fixed fields declared above are objects that MUST use keys that match the regular expression: `^[a-zA-Z0-9\.\-_]+$`.
> 
> Field Name Examples:
> 
> * `User`
> * `User_1`
> * `User_Name`
> * `user-name`
> * `my.org.User`

So the backtick is an invalid character by the spec.
A quick fix would be to replace this with an underscore. We can update the settings:

```csharp
services.AddAsyncApiSchemaGeneration(options =>
{
    options.SchemaIdSelector = type => new CamelCaseNamingStrategy().GetPropertyName(type.Name, false).Replace("`", "_");
    // ...
}
```

However, it's a bit unfortunate that the name of a generic type is `"generic_1"`, the type name is not exactly useful.
In your example `"nullable_1"` could be a nullable *anything*.
It would be more useful to generate a name for the type, like `nullableOfFoo`, or `listOfBar`, or `dictionaryOfStringAndFoo`.

I will update the `SchemaIdSelector` to handle generics better by default 👍

---

Also, we should probably be handling Nullable differently to other generics...
Json Schema allows for `{ "type": "null" }`, see: https://json-schema.org/understanding-json-schema/reference/null.html
Which means we could detect `System.Nullable<T>` and set the property as `{ "type": [ "T", "null" ] }`. I will investigate this.

**Comment by claorden** [comment]
As said in the other issue, that was fast! Thank you for taking a look into this. Please let me know if I can be of any help.

**Comment by tehmantra** [comment]
Fixed in v0.2.0: https://www.nuget.org/packages/Saunter/0.2.0
